tests/ui/job-view/Filtering_test.jsx

```diff
@@ -0,0 +1,275 @@
+                  comments: 'didathing',
+                },
+              ],
+            },
+          ],
+        },
+      );
+    });
+
+    test('should have 1 push', async () => {
+      const { getAllByText, getByTestId } = render(<App />);
+      await waitForElement(() => getAllByText('reviewbot'));
+      setUrlParam('author', 'reviewbot');
+      await waitForElementToBeRemoved(() => getByTestId('push-511138'));
+
+      const filteredPushes = await waitForElement(() =>
+        getAllByText('View Tests'),
+      );
+      expect(filteredPushes).toHaveLength(1);
+
+      setUrlParam('author', null);
+      await waitForElement(() => getAllByText('jarilvalenciano@gmail.com'));
+      const unFilteredPushes = await waitForElement(() =>
+        getAllByText('View Tests'),
+      );
+      expect(unFilteredPushes).toHaveLength(10);
+    });
+  });
+
+  describe('by failure result', () => {
+    beforeEach(() => {
+      // polling..  Return nothing...
+      fetchMock.get(
+        `begin:${getProjectUrl(
+          '/push/?full=true&count=11&push_timestamp__lte=',
+          'autoland',
+        )}`,
+        {
+          results: [],
+        },
+      );
+    });
+
+    test('should have 20 failures', async () => {
+      const { getAllByText, getByTitle, findAllByText } = render(<App />);
+      await waitForElement(() => findAllByText('B'));
+      const unclassifiedOnlyButton = getByTitle(
+        'Loaded failures / toggle filtering for unclassified failures',
+      );
+      fireEvent.click(unclassifiedOnlyButton);
+
+      await waitForElementToBeRemoved(() => getAllByText('yaml'));
+
+      // The api returns the same joblist for each push.
+      // 10 pushes with 2 failures each, but only 1 unclassified.
+      expect(jobCount()).toBe(10);
+
+      // undo the filtering and make sure we see all the jobs again
+      fireEvent.click(unclassifiedOnlyButton);
+      await waitForElement(() => getAllByText('yaml'));
+      expect(jobCount()).toBe(40);
+    });
+  });
+
+  describe('by keywords', () => {
+    beforeAll(() => {
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/', 'autoland'),
+        Object.values(jobMap)[0],
+      );
+      fetchMock.get(
+        getProjectUrl('/job-log-url/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/performance/data/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(getApiUrl('/jobdetail/?job_id=259537372'), { results: [] });
+      fetchMock.get(getProjectUrl('/note/?job_id=259537372', 'autoland'), []);
+      fetchMock.get(
+        getProjectUrl('/bug-job-map/?job_id=259537372', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/bug_suggestions/', 'autoland'),
+        [],
+      );
+      fetchMock.get(
+        getProjectUrl('/jobs/259537372/text_log_steps/', 'autoland'),
+        [],
+      );
```

Comment:
I'm quite surprised how many APIs need to be hit for a single job.

Reply:
Yeah, we have an old bug to convert it to a single api to get everything.  Possibly with GraphQL.  Just never hit the top of the priority list.